test/jdk/javax/swing/JTable/JTableRightOrientationTest.java

```diff
@@ -0,0 +1,210 @@
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    frame = new JFrame("Test JTable");
+                    JPanel panel = new JPanel(new GridBagLayout());
+                    frame.setContentPane(panel);
+                    customTableObj = new CustomTable();
+                    panel.add(new JScrollPane(customTableObj.table),
+                            new GridBagConstraints(0, 0, -1, -1, 1.0, 1.0,
+                                    GridBagConstraints.PAGE_START, GridBagConstraints.BOTH,
+                                    new Insets(2, 2, 2, 2), 0, 0));
+                    frame.pack();
+                    frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+                    frame.applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+                SwingUtilities.invokeAndWait(() -> {
+                    allColumnWidths = 0;
+                    for (int i = 0; i < customTableObj.table.getColumnCount(); i++) {
+                        allColumnWidths += customTableObj.table.getTableHeader().getColumnModel()
+                                .getColumn(i)
+                                .getWidth();
+                    }
+                    tableLocation = customTableObj.table.getLocationOnScreen();
+                    tableSize = customTableObj.table.getSize();
+                });
+
+                BufferedImage bufferedImage = robot.createScreenCapture(
+                        new Rectangle(tableLocation.x,
+                                tableLocation.y,
+                                (int)tableSize.getWidth() - allColumnWidths,
+                                (int)tableSize.getHeight()));
+                int expectedRGB = bufferedImage.getRGB(0, 0);
+                for (int x = 0; x < bufferedImage.getWidth(); x++) {
+                    for (int y = 0; y < bufferedImage.getHeight(); y++) {
+                        if (expectedRGB != bufferedImage.getRGB(x, y)) {
+                            saveImage(bufferedImage, "failureImage.png");
+                            throw new RuntimeException("Test Failed at <" + x + ", " + y + ">");
+                        }
+                    }
+                }
+
+                System.out.println("Test Passed!");
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+            robot.waitForIdle();
+            robot.delay(200);
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
```

Comment:
```suggestion
            ImageIO.write(image, "png", new File("failureImage.png"));
```
Perhaps, you can inline the file name… It never changes.